src/shared/sam/cli/samCliInvoker.ts

```diff
@@ -53,31 +53,38 @@ export class DefaultSamCliProcessInvoker implements SamCliProcessInvoker {
 
     public async invoke(options?: SamCliProcessInvokeOptions): Promise<ChildProcessResult> {
         const invokeOptions = makeRequiredSamCliProcessInvokeOptions(options)
-        const logger = getLogger()
+        const logging = options?.logging !== false
```

Comment:
It's kinda strange that `logging: true`  === `logging: undefined` in this case. Maybe swap to `logging` to `silent` in `options` and remove this statement.

Reply:
Isn't it equivalent to a "default parameter"? Only we cannot define the default at the declartion, in this case.

Reply:
It is (hence why I'm not blocking the approval) but it's kinda goofy that a truthy and a non-truthy value mean the same thing. Basically, would people down the line feel compelled to put `logging: true` in every SAM invocation?

Reply:
> but it's kinda goofy that a truthy and a non-truthy value mean the same thing

But that is always the case with all default parameters. Only here the implementation makes us question it, but the interface is ideal IMO. Twisting the interface/API to make the impl more comfortable, is a worse tradeoff, here.

> would people down the line feel compelled to put `logging: true` in every SAM invocation?

I added a docstring mentioning "default: true".

In general: if we are going to use maps to group parameters, it would be a high cost to always have to contort the interface so that we never default missing parameters to "true"...